packages/dds/tree/src/test/domains/json/benchmarks.ts

```diff
@@ -0,0 +1,39 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { fail, strict as assert } from "assert";
+import { ITreeCursor, reduceField, TreeNavigationResult } from "../../../forest";
+import { EmptyKey, FieldKey } from "../../../tree";
+
+export function sum(cursor: ITreeCursor): number {
+    let total = 0;
+    const value = cursor.value;
+    if (typeof value === "number") {
+        total += value;
+    }
+    for (const key of cursor.keys) {
+        total += reduceField(cursor, key, total, sum);
+    }
+    return total;
+}
+
+export function mahattanPerimeter(
```

Comment:
Looking at this implementation, there is a bit more non-cursor overhead than I'd really like.

I think if instead of doing perimeter we did average location, we could avoid the undefined test, `abs` calls and, the array allocation and subtractions.